Engine/Entities/ResourceEntity.cs

```diff
@@ -13,9 +14,28 @@ public ResourceEntity(World world, int quantity, Resource resource) : base(world
             Type = resource;
         }
 
-        public int Quantity { get; }
+        public int Quantity { get; private set; }
         public Resource Type { get; }
 
+        public void AddResource(int quantity)
+        {
+            Quantity += quantity;
+        }
+
+        public int RemoveResource(int quantity)
+        {
+            int newValue = Quantity - quantity;
+            if (newValue <= 0)
+            {
+                quantity = Quantity;
```

Comment:
Instead of modifying the variable passed as a parameter to RemoveResource, you could just change newValue to be == 0 and check new value afterwards? (Quanitity = newValue on line 33)